internal/configs/config_params.go

```diff
@@ -179,6 +179,7 @@ func NewDefaultConfigParams(isPlus bool) *ConfigParams {
 		MainKeepaliveRequests:           100,
 		VariablesHashBucketSize:         256,
 		VariablesHashMaxSize:            1024,
+		MainAppProtectDosLogFormat:      ", vs_name_al=$app_protect_dos_vs_name, ip=$remote_addr, tls_fp=$app_protect_dos_tls_fp, outcome=$app_protect_dos_outcome, reason=$app_protect_dos_outcome_reason, policy_name=$app_protect_dos_policy_name, dos_version=$app_protect_dos_version, ip_tls=$remote_addr:$app_protect_dos_tls_fp, ",
```

Comment:
hm. it is unusual  for a log format to start with a `,`. Is it intentional?

also note that in the nginx template, we should always write as mentioned here https://github.com/nginxinc/kubernetes-ingress/pull/2004#discussion_r717081666 , so it should be:
```
    {{- if .AppProtectDosLoadModule}}
    log_format log_dos '{{.AppProtectDosLogFormat}}';
    {{- end}}
```

Note that you can also define the default log format in the template, similar to how it is done for the main format:
```
    {{if .LogFormat -}}
    log_format  main {{if .LogFormatEscaping}}escape={{ .LogFormatEscaping }} {{end}}
                     {{range $i, $value := .LogFormat -}}
                     {{with $value}}'{{if $i}} {{end}}{{$value}}'
                     {{end}}{{end}};
    {{- else -}}
    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';
    {{- end}}
```
That's what I suggest to do.

Additionally, for consistency with the main log format, I suggest support configuring log format across multiple lines like this:
```
  log-format: |
    {
     "remote_addr": "$remote_addr",
     "remote_user": "$remote_user",
     "request": "$request"
    }
```
For that the following loop is needed:
```
                     {{range $i, $value := .AppProtectDosLogFormat -}}
                     {{with $value}}'{{if $i}} {{end}}{{$value}}'
                     {{end}}{{end}};
```

A configurable configmap key for escaping can be added later if requested by users.

Reply:
>  log format to start with a ,. Is it intentional? 
This format is used for the Kibana that our tests support

> configuring log format across multiple lines
Done

> A configurable configmap key for escaping
Done